src/optionsutils.nim

```diff
@@ -497,3 +497,39 @@ template optCmp*(self, cmp, value: untyped): untyped =
       if `cmp`(unsafeGet(a), unsafeGet(b)):
         return a
   )()
+
+import sequtils
+
+macro withSome*(procDef: untyped): untyped =
+  ##Early exit if the any of option parameters passed are none.
+  ##Also shadows the parameters to their internal type.
+  doAssert procDef.kind == nnkProcDef, "This macro only works on procedure definitions."
+  let 
+    identDefs = procDef[3] #All parameter names
+    stmtList = procDef.body #body
+  for def in identDefs:
+    let bracket = def.findChild(it.kind == nnkBracketExpr)
+    if bracket != nil and $bracket[0] == "Option":
+      #Get all defined variables here so we can check them later
+      let idents = def.filterIt(it.kind == nnkIdent)
```

Comment:
This will break for `x: Option[int] = someConst` as `someConst` would be an nnkIdent as well and pass the filter.

Reply:
Im uncertain if you're right, as my updated commit will show it works with a defined constant. So unless I missunderstand it's a non issue.